.github/workflows/test.yml

```diff
@@ -0,0 +1,13 @@
+name: Unit tests
+on: [push]
```

Comment:
You probably want this to trigger on all pull requests, but only on pushes to the `main` branch. Otherwise a push to an open PR will trigger the workflow to run twice

Reply:
I think it would only run twice if I'd specified `[push, pull_request]` here? I tried making a PR and only had one test run occur: https://github.com/anoadragon453/matrix-content-scanner/pull/1

Reply:
Ah, what I mean is that you probably want it to run on `pull_request` as well, since otherwise I don't think PRs from external forks wouldn't trigger the workflow?

But I could be wrong. Should we merge it and see?

Reply:
Aha, I didn't think about how `pull` wouldn't cover external repositories. I actually cribbed this behaviour from Sygnal: https://github.com/matrix-org/sygnal/blob/main/.github/workflows/pipeline.yml#L2

And it indeed looks like tests aren't being run on contributor PRs: https://github.com/matrix-org/sygnal/pull/227 I'll PR a fix to Sygnal as well as update this PR.

Reply:
I think ideally I'd want to have tests run on a push to any branch on the repository, even if it's not part of a pull request. However, I'm not sure how to achieve that automatically without also triggering two jobs to run when pushing to a PR branch.

It also doesn't look like it's possible to manually run a workflow on an arbitrary branch alas.